spartan/scripts/prepare_sepolia_accounts.sh

```diff
@@ -56,35 +79,67 @@ if ! command -v yq &>/dev/null; then
   chmod +x /usr/local/bin/yq
 fi
 
-# Create a new mnemonic with the required number of accounts
+# Create a new mnemonic
 echo "Creating mnemonic..."
-cast wallet new-mnemonic --accounts "$num_accounts" --json >"$tmp_filename"
+cast wallet new-mnemonic --json >"$tmp_filename"
 MNEMONIC=$(jq -r '.mnemonic' "$tmp_filename")
-ADDRESSES=$(jq -r '.accounts[].address' "$tmp_filename")
 
-# Convert ETH to wei
-wei_amount=$(cast to-wei "$eth_amount" ether)
+echo "MNEMONIC: $MNEMONIC"
 
-# Get current gas price and add 25% buffer
-echo "Getting gas price..."
-gas_price=$(cast gas-price --rpc-url "$ETHEREUM_HOST")
-gas_price=$((gas_price * 125 / 100)) # Add 25% to gas price
+# Cast has a limit of 255 accounts per mnemonic command
+# We'll need to derive accounts in batches
+echo "Deriving $total_accounts accounts from mnemonic..."
+
+# Function to derive address at a specific index
+derive_address() {
+  local index=$1
+  local mnemonic=$2
+  cast wallet address --mnemonic "$mnemonic" --mnemonic-index "$index"
+}
+
+# Build an array of indices to fund
+declare -a indices_to_fund
 
-# Build 'calls' string in the format:
-# [(0xADDR,false,wei_amount,0x),(0xADDR2,false,wei_amount,0x)]
+# Add validator indices
+for ((i = 0; i < num_validators; i++)); do
+  indices_to_fund+=($((validator_key_index_start + i)))
+done
+
+# Add prover indices
+for ((i = 0; i < num_provers; i++)); do
+  indices_to_fund+=($((prover_key_index_start + i)))
+done
+
+# Add bot indices if enabled
+if [ "$bot_enabled" = "true" ]; then
+  for ((i = 0; i < num_bots; i++)); do
+    indices_to_fund+=($((bot_key_index_start + i)))
+  done
+fi
+
+# Get the addresses to fund
 calls="["
-for addr in $ADDRESSES; do
-  calls+="(${addr},false,${wei_amount},0x),"
+num_accounts_to_fund=${#indices_to_fund[@]}
+
+echo "Deriving addresses for $num_accounts_to_fund accounts to fund..."
+for index in "${indices_to_fund[@]}"; do
+  address=$(derive_address "$index" "$MNEMONIC")
+  calls+="(${address},false,${wei_amount},0x),"
 done
 calls=${calls%,}
 calls+="]"
 
-# Total value = wei_amount * num_accounts
-total_value=$(echo "$wei_amount * $num_accounts" | bc)
+# Get current gas price and add 25% buffer
+echo "Getting gas price..."
+gas_price=$(cast gas-price --rpc-url "$ETHEREUM_HOST")
+gas_price=$((gas_price * 125 / 100)) # Add 25% to gas price
+
+# Total value = wei_amount * num_accounts_to_fund
+total_value=$(echo "$wei_amount * $num_accounts_to_fund" | bc)
```

Comment:
I'd add a check that `total_value` is not over a stupidly large amount, just in case, so we don't accidentally empty out our funding account